akka-http/src/main/resources/reference.conf

```diff
@@ -37,6 +37,13 @@ akka.http.routing {
   # for an entity data stream.
   decode-max-bytes-per-chunk = 1m
 
+  # Maximum content length after applying a decoding directive. When the directive
+  # decompresses, for example, an entity compressed with gzip, the resulting stream can be much
+  # bigger than the max-content-length. Like with max-content-length, this is not necessarilly a
+  # problem when consuming the entity in a streaming fashion, but does risk high memory use
+  # when the entity is made strict or marshalled into an in-memory object.
+  decode-max-size = 8m
```

Comment:
Should it default to the value of max-content-size to try to not break applications for users that increased that?

Reply:
I'm not sure that is possible: if `max-content-size` is updated in `application.conf`, AFAICS referring to `max-content-size` from this place in the `reference.conf` still wouldn't point to the new value if I'm reading https://github.com/lightbend/config#note-about-resolving-substitutions-in-referenceconf-and-applicationconf right.

Reply:
Yes, you are right. It wouldn't work.

Reply:
What about solving that when reading it then, leave this undefined/empty and use the other one unless this has a specific value?

Reply:
That could work - leaving it undefined would confuse config-checker, but we could introduce another 'magic value' here and handle it specially in code. Do we have any other places where we do something similar?

Reply:
We do break applications anyway where uncompressed data is bigger than either the new limit or - if we go with this proposal - whatever `max-content-length` was set to. So, I'd vote for keeping it simple and explicit as it is here.

The good thing is, that `withSizeLimit` now works for overriding both limits.